sympy/stats/rv_interface.py

```diff
@@ -208,6 +209,59 @@ def skewness(X, condition=None, **kwargs):
     """
     return smoment(X, 3, condition, **kwargs)
 
+def quantile(X, p):
+    r"""
+    Return the :math:`p^{th}` order quantile of a probability distribution.
+
+    Quantile is defined as the value at which the probability of the random
+    variable is less than or equal to the given probability.
+
+    ..math::
+        Q(p) = inf{x \in (-\infty, \infty) such that p < F(x)}
+
+    Examples
+    ========
+
+    >>> from sympy.stats import quantile, Die, Exponential
+    >>> from sympy import Symbol, pprint
+    >>> p = Symbol("p", positive=True)
+
+    >>> l = Symbol("lambda", positive=True)
+    >>> X = Exponential("x", l)
+    >>> pprint(quantile(X, p), use_unicode=False)
+               -log(1 - p)
+    [0, oo) n {------------}
+                  lambda
+
+    >>> D = Die("d", 6)
+    >>> pprint(quantile(D, p), use_unicode=False)
+    /1  for p <= 1/6
+    |
+    |2  for p <= 1/3
+    |
+    |3  for p <= 1/2
+    <
+    |4  for p <= 2/3
+    |
+    |5  for p <= 5/6
+    |
+    \6   for p < 1
+    """
+    _value_check(p > 0, "The order p must be positive.")
+    _value_check(p < 1, "The order p must be less than 1.")
```

Comment:
Yes, reference: https://en.wikipedia.org/wiki/Quantile

Reply:
It makes sense to call `quantile(X)(0)` or `quantile(X)(1)`.

Reply:
Actually, p=1 has no meaning in reference in case of quantile (except finite cases). Consider exponential distribution. The quantile is denoted as `(-log(1-p))/rate`. For p = 1, numerator has log(0). In this case, however, p=0 seems valid.
This result comes directly from the distribution function. For exponential distribution, the domain is [0, oo). That is why p=0 gives results in this case. For most of the distributions, the domain is (-oo, +oo). In such cases quantile doen't exist for both p=0 and p=1.
For discrete cases, p=0 always exists but it is not accepted worldwide.

Reply:
It will return oo.

Reply:
Yes, that would be right for limiting case. But explicitly p=1 has no meaning. Perhaps, one can use `limit` for such cases

Reply:
Maybe I should add explaination for such limiting cases in docs and illustrate the same by adding examples to the docs.

Reply:
Why all this hussle? We support returning infinity, look at the tangent function and pass pi/2. Limits are a separate concept.

Reply:
Ok, I'll do the changes